source/Engine.cpp

```diff
@@ -483,7 +483,10 @@ void Engine::Step(bool isActive)
 	if(flagship && flagship->GetTargetStellar() && !isJumping)
 	{
 		const StellarObject *object = flagship->GetTargetStellar();
```

Comment:
this could be `const Planet *planet = flagship->GetTargetStellar()->GetPlanet();` since this block is conditioned on the stellar object existing.
Then you'd be able to drop the extra `object->GetPlanet()` check below.